### PR TITLE
[MU4] Audio redesigned using synthesizer

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: framework/midi/internal/sequencer.cpp [hunk 1]
```diff
@@ -363,15 +346,15 @@ bool Sequencer::sendChanEvents(const Channel& chan, uint32_t ticks)
             return ret;
         }
 
-        if (state.muted || event.type == MIDI_EOT || event.type == META_TEMPO) {
+        if (!m_isPlayTickSet) {
+            m_playTick = event.tick;
+            m_isPlayTickSet = true;
+        }
+
+        if (state.muted || event.type == MIDI_EOT || event.type == META_TEMPO || event.type == ME_TICK1 || event.type == ME_TICK2) {
```
Comment (RomanPudashkin) [inline framework/midi/internal/sequencer.cpp:354]: Better to use a set of types here, something like that:

QSet allowedTypes {  ... }

if (!state.muted && allowedTypes.contains(type)) { ... }

File: framework/audio/iaudioengine.h [hunk 2]
```diff
@@ -51,6 +51,7 @@ class IAudioEngine : MODULE_EXPORT_INTERFACE
     virtual Ret init() = 0;
     virtual void deinit() = 0;
     virtual bool isInited() const = 0;
+    virtual async::Channel<bool> initChanged() const = 0;
```
Comment (RomanPudashkin) [inline framework/audio/iaudioengine.h:54]: Can you explain to me please, what this notification is needed for?

File: framework/midi/internal/sequencer.cpp (hunk 1 above)
Comment (igorkorsukov) [reply framework/midi/internal/sequencer.cpp:354]: I also thought about this and decided that if at least one more type is added (and ceases to fit on one line), then I will change it into a set

File: framework/midi/internal/synthesizersetup.cpp [hunk 3]
```diff
@@ -16,18 +16,32 @@
...
-
-#include "sffproviderlocalfile.h"
+#include "synthesizersetup.h"
 
 #include "log.h"
 
 using namespace mu::midi;
 
-void SFFProviderLocalFile::loadSF(const midi::Programs& programs, const OnLoading& onloading, const OnLoaded& onloaded)
+void SynthesizerSetup::setup()
```
Comment (RomanPudashkin) [inline framework/midi/internal/synthesizersetup.cpp:25]: I propose to add a callback to AudioEngine::init and make setup of the synthesizer in it

File: framework/audio/iaudioengine.h (hunk 2 above)
Comment (igorkorsukov) [reply framework/audio/iaudioengine.h:54]: To initialize the synthesizer, we need a sample rate. The sample rate can be obtained only after initializing the audio engine. So we can initialize the synthesizer only after initializing the audio engine. If the audio engine is not initialized at the moment, I need to subscribe and wait for it to be initialized.
https://github.com/musescore/MuseScore/pull/6399/files#diff-57c3ba1eac6f3d5a4de940828b780ad2R38

File: framework/midi/internal/synthesizersetup.cpp (hunk 3 above)
Comment (igorkorsukov) [reply framework/midi/internal/synthesizersetup.cpp:25]: The audio engine is in one module that knows nothing about the synthesizer. The synthesizer is in another module, which does not call the audio engine init

**Pull request merged**